lib/jekyll/regenerator.rb

```diff
@@ -164,6 +164,9 @@ def read_metadata
           Marshal.load(content)
         rescue TypeError
           SafeYAML.load(content)
+        rescue ArgumentError => e
```

Comment:
This makes it so that we don't crash the whole process and leave the user wondering if we ever fail to read the metadata for some reason (for example because it crashed because of bugs like the above).